src/settings/components/ExternalDataSourcesConfig/ExternalDataSourcesEdit.js

```diff
@@ -22,6 +23,16 @@ export default class ExternalDataSourcesEdit extends React.Component {
     onSave: PropTypes.func.isRequired,
   }
 
+  validateUniqueName = (value, allValues) => {
+    const { externalKbs } = allValues;
+    const uniqueNameSources = externalKbs.filter(externalKb => externalKb.name.toLowerCase() === value.toLowerCase());
```

Comment:
If you wanted to show off your Array skillz you could even use `Array.some` 😆  

But yeah, it doesn't really make a difference here because the short-circuit evaluation probably won't apply to anyone since iterating over the entire list of KBs doesn't make a difference performance-wise.